docs/js-core-3/module-systems/lesson.md

```diff
@@ -0,0 +1,399 @@
+- Then we have the `from` keyword which tells JS where we want to import the code from
+- Finally we use another file path to describe where the file is
+
+### The `export` keyword
+
+Exporting using the ESM system is also very similar to CJS. We just swap `module.exports` with the `export` keyword:
+
+```js
+function findWidget(id) {
+  // ...
+}
+
+export default findWidget;
+```
+
+Similar to CJS, we've defined a `findWidget` function just like any other function. The important bit here is the `export` keyword, which tells JS that we want to export the `findWidget` reference. Remember that when we define a function like this, we also create a reference that can be used like any other variable. We can now import the `findWidget` function into the rest of our app.
+
+You might notice we also added another keyword, `default` when exporting `findWidget`. This tells the module system that this is the "main" export from this file. If you only have one thing that you'd like to export, then we tend to make it a `default` export.
+
+Sometimes you might see the export shortened to this:
+
+```js
+export default function findWidget(id) {
+  // ...
+}
+```
+
+Here, we're defining the function and exporting it (as the `default`) in same line.
+
+If we want to export more than "main" thing from a file, then we can leave off the `default` keyword:
+
+```js
+export function findWidget() {
+  // ...
+}
+
+export function findAllWidgets() {
+  // ...
+}
+```
+
+In this code, we've marked both the `findWidget` and `findAllWidgets` with the `export` keyword, which means they'll both be available as exported functions.
+
+Then when we import this file, we can use some slightly different syntax with the `import` keyword:
+
+```js
+import { findWidget, findAllWidgets } from "./widget-finder";
+```
+
+We now have access to both the `findWidget` and `findAllWidget` functions.
+
+:::tip
+You can think of this `import` syntax like destructuring an object.You can imagine that we are importing an object from the `widget-finder` file, then destructuring the exported references out of that object.
+:::
+
+## Using module systems in your code
+
+Unfortunately, deciding which module system to use is currently quite complicated. Both systems are actively being used in different apps, but often they are **incompatible** with each other 😕
+
+The main reason for this is that ESM was created after CJS and so a lot of older code has not been moved away from CJS. To make things worse, in some cases, it is currently impossible to move code to ESM.
+
+For example, when running code in Node, then CJS will tend to work much better than ESM. Similarly when running code in a browser, then CJS will not work at all, but ESM will work.
+
+:::tip
+The rule of thumb is when your code will run in Node then you should use the CJS module system. And when your code will run in the browser, use the ESM module system.
```

Comment:
what about react? (and where does that leave us with helping students with TDD, especially if we'd like them to spend more time in the browser?)